### ISSUE TITLE
[REC] ezcheck

### ISSUE THREAD

**Issue by edwinjhlee** [x-cmd/x-cmd#109]
## Website

https://github.com/metaphorme/ezcheck

## Screenshot

<img width="782" alt="image" src="https://github.com/user-attachments/assets/b1823183-9634-4b38-8b25-27c8d6aa1ecf">

## Ready to be included in [x-cmd pkg](https://x-cmd.com/pkg) ?

YES or N

## How to install

Download from https://github.com/Metaphorme/ezcheck/releases/tag/0.1.3

## TLDR

```bash
# Provide a TLDR like example to help us understand
# Don't hesitate ~ The more the better !!!

# Auto detect hash algorithm
ezcheck compare -f image.jpg -c bebc102992450c68e5543383889e27c9

# ezcheck check [ALGORITHM (Leave blank to automatically detect algorithm)] -c check-file
ezcheck check -c sha256sum.txt 

# ezcheck calculate [ALGORITHM (Default SHA256)]
ezcheck calculate sha256 -t "Hello"
```

## Highlights

ezcheck(easy check) is an ergonomic, standard-output command-line tool for calculating, comparing, and verifying the hash of strings and files.

- Discuss the advantages of this tool: performant, cross-platform, consistent
- Stories about this tool: https://github.com/Metaphorme/ezcheck/issues/1
- Discuss the disadvantages of this tool: could be more powerful

**Comment by Metaphorme** [comment]
Thanks for including ezcheck into x-cmd!

I released new version [0.1.4](https://github.com/Metaphorme/ezcheck/releases/tag/0.1.4) yesterday and proved some new features for ezcheck. Here are the new features and more information about ezcheck:

## How to install

* Download from [0.1.4](https://github.com/Metaphorme/ezcheck/releases/tag/0.1.4) with the file name including `mix_backend`, which provides both fastest speed and maximum compatibility.

  Build targets in release: 
    * x86_64-unknown-linux-gnu
    * x86_64-unknown-linux-musl
    * aarch64-unknown-linux-gnu
    * aarch64-unknown-linux-musl
    * i686-unknown-linux-gnu
    * loongarch64-unknown-linux-gnu
    * loongarch64-unknown-linux-musl
    * x86_64-pc-windows-msvc
    * x86_64-pc-windows-gnu
    * i686-pc-windows-msvc
    * i686-pc-windows-gnu
    * aarch64-pc-windows-msvc
    * aarch64-apple-darwin
    * x86_64-apple-darwin

By the way, here are the steps to [build from source](https://github.com/Metaphorme/ezcheck?tab=readme-ov-file#build-from-source) and [test](https://github.com/Metaphorme/ezcheck?tab=readme-ov-file#run-tests).

## TLDR

```
# Provide a TLDR like example to help us understand
# Don't hesitate ~ The more the better !!!

# Supported algorithms:
MD2, MD4, MD5, SHA1, SHA224, SHA256, SHA384, SHA512, SHA512/256

# Calculate hash for a file or text

## Usage: ezcheck calculate|c [ALGORITHM (Default SHA256)] (-f file(s)/"-" for standard input | -t text)
ezcheck calculate sha256 -f image.jpg

## Or use the shortcut (c for calculate)
ezcheck c sha256 -t "Hello"

## Leave algorithm blank to use sha256 as the default
ezcheck c -f image.jpg

## Read from standard input
cat image.jpg | ezcheck c -f -

# Compare with given hash

## Usage: ezcheck compare|m [ALGORITHM (Leave blank to automatically detect algorithm)] (-f file/"-" for standard input | -t text) -c hash
ezcheck compare sha256 -f image.jpg -c 4c03795a6bca220a68eae7c4f136d6247d58671e074bccd58a3b9989da55f56f

## Or use the shortcut (m for compare)
ezcheck m sha256 -t "Hello" -c 085f8db32271fe25f561a6fc938b2e264306ec304eda518007d1764826381969

## Leave algorithm blank to automatically detect algorithm
ezcheck m -f image.jpg -c bebc102992450c68e5543383889e27c9

# Check with given shasum file

## shasum file looks like:
00691413c731ee37f551bfaca6a34b8443b3e85d7c0816a6fe90aa8fc8eaec95  滕王阁序.txt
4c03795a6bca220a68eae7c4f136d6247d58671e074bccd58a3b9989da55f56f *image.jpg

## Usage: ezcheck check|k [ALGORITHM (Leave blank to automatically detect algorithm)] -c check-file
ezcheck check sha256 -c sha256sum.txt 

## Or use the shortcut (k for check)
ezcheck k sha256 -c sha256sum.txt 

## Leave algorithm blank to automatically detect algorithm
ezcheck k -c sha256sum.txt 
```

**Comment by lunrenyi** [comment]
ezcheck has been added to the package.
```sh
x pkg update
x env use ezcheck
```

**Comment by Metaphorme** [comment]
Thanks!
I'm so glad to see that ezcheck has been included into x-cmd package and install! So appreciate for your great work!

**Comment by Metaphorme** [comment]
Hi there!
ezcheck had released version [0.1.5](https://github.com/Metaphorme/ezcheck/releases/tag/0.1.5).
Comparing with the privious version, 0.1.5 has no change in code, but a lot of additions and fitness of documents. crates.io doesn't allow to change anything already uploaded, as a result, I released the new version.
